tensorflow_addons/seq2seq/decoder.py

```diff
@@ -253,16 +257,16 @@ def tracks_own_finished(self):
 
 
 def dynamic_decode(
-    decoder,
-    output_time_major=False,
-    impute_finished=False,
-    maximum_iterations=None,
-    parallel_iterations=32,
-    swap_memory=False,
-    training=None,
-    scope=None,
+    decoder: Decoder,
```

Comment:
`decoder` can also be a `BaseDecoder`.

Reply:
Thanks @guillaumekln. In docstring mentioning like this, `decoder: A `Decoder` instance.` Can you please let me know more?

Reply:
See the current type check: https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/decoder.py#L301

Reply:
Got it, thanks @guillaumekln.